logicrunner/execution_archive.go

```diff
@@ -0,0 +1,142 @@
+//
+// Copyright 2019 Insolar Technologies GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package logicrunner
+
+import (
+	"context"
+	"sync"
+
+	"github.com/insolar/insolar/insolar"
+	"github.com/insolar/insolar/insolar/jet"
+	"github.com/insolar/insolar/insolar/message"
+	"github.com/insolar/insolar/insolar/record"
+)
+
+type Archiver interface {
+	Archive(transcript *Transcript)
+}
+
+//go:generate minimock -i github.com/insolar/insolar/logicrunner.ExecutionArchive -o ./ -s _mock.go
+type ExecutionArchive interface {
+	Archive(transcript *Transcript)
+	Done(transcript *Transcript) bool
+
+	IsEmpty() bool
+	OnPulse(ctx context.Context) []insolar.Message
+	FindRequestLoop(ctx context.Context, apiRequestID string) bool
```

Comment:
strange to see this here

Reply:
execution archive stores all information about running tasks, so it should work as "loop detection part", now.